clientd3d/d3drender.c

```diff
@@ -186,6 +184,31 @@ void SetZBias(LPDIRECT3DDEVICE9 device, int z_bias) {
                                    *((DWORD *) &bias));
 }
 
+bool isFogEnabled()
+{
+	return gD3DDriverProfile.bFogEnable;
+}
+
+long getShadeAmount()
+{
+	return shade_amount;
+}
+
+const Vector3D& getSunVector()
+{
+	return sun_vect;
+}
+
+const std::pair<int,int> getPlayerPosition()

```

Comment:
Let's not put this here.  If we want to remove the player extern, a function like this should go in a more central file, not in the 3D renderer.  Also, a function like this shouldn't allocate memory (std::pair).  There are dozens and dozens of player externs, so I suggest fixing this in a future PR.